rclcpp/include/rclcpp/publisher_base.hpp

```diff
@@ -203,6 +205,46 @@ class PublisherBase : public std::enable_shared_from_this<PublisherBase>
   std::vector<rclcpp::NetworkFlowEndpoint>
   get_network_flow_endpoints() const;
 
+  /// Wait until all published messages are acknowledged or until the specified timeout elapses.
+  /**
+   * This method waits until all published messages are acknowledged by all matching
+   * subscriptions or the given timeout elapses.
+   *
+   * If the timeout is negative then this method will block indefinitely until all published
+   * message data are acknowledged.
+   * If the timeout is zero then this method will not block, it will check if all published
+   * messages were acknowledged and return immediately.
+   * If the timeout is greater than zero, this method will wait until all published messages are
+   * acknowledged or the timeout elapses.
+   *
+   * This method only waits for acknowledgments if the publisher's QoS profile is RELIABLE.
+   * Otherwise this method will immediately return `true`.
+   *
+   * \param[in] timeout the duration to wait for all published message data to be acknowledged.
+   * \return `true` if all published message data were acknowledged before the given timeout
```

Comment:
```suggestion
   * \param[in] timeout the duration to wait for all published messages to be acknowledged.
   * \return `true` if all published messages were acknowledged before the given timeout
```

Reply:
Okay.  
I will check if the contents has the similar  problems and resolve this type of problem once and for all.